File: auto-docs/executables/node/YIGnBu_heatmap.js
var plotly = require('plotly')('theengineear', 'o9zlr0hy6z')

var unpack = require('ndarray-unpack');
var zeros = require('zeros');
var fill = require('ndarray-fill');

var size = 50;
var z = unpack(fill(zeros([size, size]), function(c,r) {
  return Math.sqrt((r*c)/(size*size));
}));
var data = [
  {
    z: z, 
    scl: "YIGnBu", 
    type: "heatmap"
  }
];
var layout = {title: "YIGnBu"};

var graph_options = {filename: "YIGnBu-heatmap", fileopt: "overwrite", layout: layout, auto_open: "false"}
plotly.plot(data, graph_options, function (err, msg) {
    console.log(msg);
});

File: auto-docs/executables/r/YIGnBu_heatmap.r
library(plotly)

p <- plotly(username='TestBot', key='r1neazxo9w')

size <- 50
z <- matrix(0, size, size)
for(r in 1:size) {
    for(c in 1:size) {
        z[r, c] <- r+c
    }
}

data <- list(
  list(
    z = z, 
    scl = "YIGnBu", 
    type = "heatmap"
  )
)
layout <- list(title = "YIGnBu")

response <- p$plotly(data, kwargs=list(layout=layout, filename="YIGnBu-heatmap", fileopt="overwrite", auto_open="FALSE"))
url <- response$url
filename <- response$filename